### PR TITLE
Fix Scheduler recurrence appointment rendering after placing into allDay panel (T831801)

### PR DESCRIPTION
**Pull request by Yarovaya**

### CODE REVIEW

File: js/ui/scheduler/ui.scheduler.subscribes.js [hunk 1]
```diff
@@ -418,14 +418,15 @@ const subscribes = {
...
+        const endDate = options.endDate;
+        const endDayHour = this._getCurrentViewOption('endDayHour');
+        const startDayHour = this._getCurrentViewOption('startDayHour');
+
+        let updatedEndDate = endDate;
 
         if(endDate.getHours() >= endDayHour) {
             updatedEndDate.setHours(endDayHour, 0, 0, 0);
-        } else if(startDayHour > 0 && (endDate.getHours() * 60 + endDate.getMinutes() < (startDayHour * 60))) {
+        } else if(!options.sameDate && startDayHour > 0 && (endDate.getHours() * 60 + endDate.getMinutes() < (startDayHour * 60))) {
```
Comment (Olexi) [inline js/ui/scheduler/ui.scheduler.subscribes.js]: This needs to be discussed later.

Comment (Yarovaya) [reply js/ui/scheduler/ui.scheduler.subscribes.js]: updateAppointmentEndDate method is needed to check whether a **long** appointment is placed in the interval [startDayHour, endDayHour]. If not, we crop the **appointment part** on the previous day. This does not need to be done for **short appointments** that do not have a part on the previous day.

Comment (Olexi) [reply js/ui/scheduler/ui.scheduler.subscribes.js]: Yes, you're absolutely right. But I think this logic should be renovated after we will implement 'FREQ=HOURLY' in our recurrence editor. What do you think about that, @Yarovaya?

Comment (Yarovaya) [reply js/ui/scheduler/ui.scheduler.subscribes.js]: After we will implement 'FREQ=HOURLY', we will have all filtered appointment parts before processing in the **updateAppointmentEndDate** method.

**Review by Olexi — DISMISSED** [review]

File: js/ui/scheduler/ui.scheduler.subscribes.js [hunk 2]
```diff
@@ -418,14 +418,15 @@ const subscribes = {
...
         const endDayHour = this._getCurrentViewOption('endDayHour');
         const startDayHour = this._getCurrentViewOption('startDayHour');
+
         let updatedEndDate = endDate;
 
         if(endDate.getHours() >= endDayHour) {
             updatedEndDate.setHours(endDayHour, 0, 0, 0);
-        } else if(startDayHour > 0 && (endDate.getHours() * 60 + endDate.getMinutes() < (startDayHour * 60))) {
+        } else if(!options.sameDate && startDayHour > 0 && (endDate.getHours() * 60 + endDate.getMinutes() < (startDayHour * 60))) {
             updatedEndDate = new Date(updatedEndDate.getTime() - toMs('day'));
```
Comment (Krijovnick) [inline js/ui/scheduler/ui.scheduler.subscribes.js:430]: Can we add test for this code?

**Review by andrewmakarov — DISMISSED** [review]

**Review by Krijovnick — APPROVED** [review]

**Review by Olexi — APPROVED** [review]

**Review by andrewmakarov — APPROVED** [review]

**Pull request merged**